### ISSUE TITLE
Use `sin_cos`

### ISSUE THREAD

**Issue by BenjaminBrienen** [rust-lang/rust-clippy#13405]
### What it does

Suggests usage of `f32::sin_cos` and `f64::sin_cos` where it would be more optimal than calling `sin` and `cos` individually.

### Advantage

sin_cos could have optimizations (although it doesn't seem to at the moment)
sin_cos supposedly exists to be more readable

### Example

Should detect:

```rs
let sin_theta = sin(theta);
let cos_theta = cos(theta);
```

And suggest:
```rs
let (sin_theta, cos_theta) = sin_cos(theta);
```

**Comment by BenjaminBrienen** [comment]
I might try implementing this after the rustweek workshop where I'll hopefully learn how to do this :)
